pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -503,3 +685,72 @@ func shouldFailWith(conditions []v2vv1alpha1.VirtualMachineImportCondition) (boo
 	}
 	return valid, message
 }
+
+func (r *ReconcileVirtualMachineImport) initProvider(instance *v2vv1alpha1.VirtualMachineImport) (provider.Provider, error) {
+	// Fetch source provider secret
+	sourceProviderSecretObj, err := r.fetchSecret(instance)
+	if err != nil {
+		return nil, err
+	}
+
+	provider, err := r.createProvider(instance)
+	if err != nil {
+		return nil, err
+	}
+
+	err = provider.Connect(sourceProviderSecretObj)
+	if err != nil {
+		return nil, err
+	}
+	defer provider.Close()
```

Comment:
It will disconnect when the function ends; we will need the ovirt client connected later.